web/client-api/src/main/java/io/deephaven/web/client/api/tree/JsTreeTable.java

```diff
@@ -214,24 +214,25 @@ public JsTreeTable(WorkerConnection workerConnection, JsWidget widget) {
                 // technically this may be set for the above two cases, but at this time we send them regardless
                 keyColumns.push(column);
             }
-            if (definition.isRollupConstituentNodeColumn()) {
-                constituentColumns.put(column.getName(), column);
-            }
             if (definition.isVisible()) {
-                columns[columns.length] = column;
-            }
-            if (definition.isRollupGroupByColumn() && !definition.isRollupConstituentNodeColumn()) {
-                groupedColumns.push(column);
-
-                if (hasConstituentColumns) {
-                    column.setConstituentType(columnDefsByName.get(true).get(definition.getName()).getType());
+                if (definition.isRollupConstituentNodeColumn()) {
+                    constituentColumns.put(column.getName(), column);
+                } else {
+                    columns[columns.length] = column;
+                    if (definition.isRollupGroupByColumn() && !definition.isRollupConstituentNodeColumn()) {
```

Comment:
No need for the `&& !definition.isRollupConstituentNodeColumn()`, as that's always true in this `else` block
```suggestion
                    if (definition.isRollupGroupByColumn()) {
```

Reply:
Thanks, also factoring out the getRollupAggregationInputColumn to only read it once.